### PR TITLE
feat(sanity): adopt `bundlePerspective` for listing documents

### PR DESCRIPTION
**Pull request by juice49**
### Description

This branch uses Content Lake to collate version documents, rather than collating them client-side. When listing/searching documents, Studio no longer fetches version documents directly. Instead, it uses the `bundlePerspective` parameter to overlay the desired version server-side.

As part of this change, global search now correctly shows list previews for the selected perspective. Documents outside the selected perspective bundle can still be found, but they are shown in a lower contrast style that matches document lists.

### What to review

- Appearance of documents in document lists when viewing different perspectives.
- Appearance of documents in global search when viewing different perspectives.

### CODE REVIEW

File: packages/sanity/src/core/studio/components/navbar/search/contexts/search/SearchProvider.tsx
Comment (pedrobonamin) [inline packages/sanity/src/core/studio/components/navbar/search/contexts/search/SearchProvider.tsx:148]: probably overkill, but it took me a bit to understand what was going on here.
Wdyt if we move this to a function that returns the params we need.

```ts
const getPerspective = (
  perspective: string | undefined,
):
  | {
      bundlePerspective: string
    }
  | {
      perspective: string | undefined
    } => {
  if (perspective?.startsWith('bundle.')) {
    const [_identifier, bundleId] = perspective.split('bundle.')
    // When querying bundles, we should use the bundleId and the drafts as the bundlePerspective e.g. [bundleId, 'drafts']
    return {bundlePerspective: [bundleId, DRAFTS_FOLDER].join(',')}
  }
  return {
    perspective: perspective,
  }
}

```

**Review by pedrobonamin — APPROVED** [review]
This looks great Ash, thanks for the great work! 
Added some suggestions, non blocking, feel free to discard them.

File: packages/sanity/src/core/studio/components/navbar/search/contexts/search/SearchProvider.tsx
Comment (pedrobonamin) [reply packages/sanity/src/core/studio/components/navbar/search/contexts/search/SearchProvider.tsx:148]: We could do the same in the text search

File: dev/test-studio/sanity.config.ts
Comment (juice49) [reply dev/test-studio/sanity.config.ts:154]: @pedrobonamin we should be able to revert it now that Radhe has setup the `test` dataset to work properly with versions 🙌. I'll test out that dataset, and if all is well, we can remove all of the workspace changes before merging this branch.

File: packages/sanity/src/core/preview/documentPreviewStore.ts
Comment (juice49) [reply packages/sanity/src/core/preview/documentPreviewStore.ts:144]: My intention was the commit name [`feat(sanity): #TEMPORARY use API vX`](https://github.com/sanity-io/sanity/pull/7415/commits/0d94937894accf4a48ae8cb2e85846f38d6f325e) would be clear enough, but adding a comment is probably a good idea 🙂.

File: packages/sanity/src/core/studio/components/navbar/search/contexts/search/SearchProvider.tsx
Comment (juice49) [reply packages/sanity/src/core/studio/components/navbar/search/contexts/search/SearchProvider.tsx:148]: I think it's a great point. It's definitely not the most intuitive code to read.

File: packages/sanity/src/core/preview/documentPreviewStore.ts
Comment (pedrobonamin) [reply packages/sanity/src/core/preview/documentPreviewStore.ts:144]: But when merging we squash so we lose that commit right?

File: dev/test-studio/sanity.config.ts
Comment (juice49) [reply dev/test-studio/sanity.config.ts]: Reverted.

File: packages/sanity/src/core/preview/documentPreviewStore.ts
Comment (juice49) [reply packages/sanity/src/core/preview/documentPreviewStore.ts:144]: I was not planning to squash this merge. I often keep things atomic and then merge every commit.

**Pull request merged**
